magda-storage-api/src/createApiRouter.ts

```diff
@@ -29,6 +29,64 @@ export default function createApiRouter(options: ApiRouterOptions) {
         return res.status(200).send("OK");
     });
 
+    /**
+     * @apiGroup Storage
+     *
+     * @api {PUT} /v0/{bucketid} Request to create a new bucket
+     *
+     * @apiDescription Creates a new bucket with a specified name. Restricted to admins only.
+     *
+     * @apiParam (Request body) {string} bucketid The name of the bucket to be created
+     *
+     * @apiSuccessExample {json} 201
+     *    {
+     *        "message":"Bucket my-bucket created successfully in unspecified-region 🎉"
+     *    }
+     *
+     * @apiSuccessExample {json} 201
+     *    {
+     *        "message": "Bucket my-bucket already exists 👍"
+     *    }
+     * @apiErrorExample {json} 500
+     *    {
+     *        "message": "Bucket creation failed. This has been logged and we are looking into this."
+     *    }
+     */
+    router.put(
+        "/:bucketid",
+        mustBeAdmin(options.authApiUrl, options.jwtSecret),
+        async function(req, res) {
+            const bucketId = req.params.bucketid;
+            if (!bucketId) {
+                return res
+                    .status(400)
+                    .send("Please specify a bucket name in the request URL.");
+            }
+
+            const encodedBucketname = encodeURIComponent(bucketId);
+            const createBucketRes = await options.objectStoreClient.createBucket(
+                encodedBucketname
+            );
+            if (createBucketRes.success) {
+                return res.status(201).send({
+                    message: createBucketRes.message
+                });
+            } else {
+                if (createBucketRes.err) {
+                    return res.status(500).send({
+                        message:
+                            "Bucket creation failed. This has been logged and we are looking into this."
+                    });
+                } else {
+                    // If bucket already exists
+                    return res.status(201).send({
+                        message: createBucketRes.message
+                    });
+                }
```

Comment:
Because you're rejecting from`createBucket` now, you'll need to put the call in a `try catch` and move the 500 status to be part of the `catch` clause.

Not throwing an error for the bucket already existing is good though, because exceptions should be exceptional :)

Reply:
Good `catch`! Fixed now